### ISSUE TITLE
SYCL :: Reduction/reduction_span_pack.cpp fails on hip/amd after pulldown

### ISSUE THREAD

**Issue by dwoodwor-intel** [intel/llvm#6409]
**Describe the bug**

`SYCL :: Reduction/reduction_span_pack.cpp` fails when run on certain HIP/AMD configurations after the pulldown in #6358, giving this output:

```
******************** TEST 'SYCL :: Reduction/reduction_span_pack.cpp' FAILED ********************
Script:
--
: 'RUN: at line 1';    /__w/llvm/llvm/toolchain/bin/clang++     -Xsycl-target-backend=amdgcn-amd-amdhsa --offload-arch=gfx1031 -fsycl -fsycl-targets=amdgcn-amd-amdhsa /__w/llvm/llvm/llvm_test_suite/SYCL/Reduction/reduction_span_pack.cpp -o /__w/llvm/llvm/build/SYCL/Reduction/Output/reduction_span_pack.cpp.tmp.out
: 'RUN: at line 2';   true /__w/llvm/llvm/build/SYCL/Reduction/Output/reduction_span_pack.cpp.tmp.out
: 'RUN: at line 3';    env SYCL_DEVICE_FILTER=ext_oneapi_hip:gpu,host  /__w/llvm/llvm/build/SYCL/Reduction/Output/reduction_span_pack.cpp.tmp.out
: 'RUN: at line 4';   true /__w/llvm/llvm/build/SYCL/Reduction/Output/reduction_span_pack.cpp.tmp.out
--
Exit Code: 16
Command Output (stdout):
--
$ ":" "RUN: at line 1"
note: command had no output on stdout or stderr
$ "/__w/llvm/llvm/toolchain/bin/clang++" "-Xsycl-target-backend=amdgcn-amd-amdhsa" "--offload-arch=gfx1031" "-fsycl" "-fsycl-targets=amdgcn-amd-amdhsa" "/__w/llvm/llvm/llvm_test_suite/SYCL/Reduction/reduction_span_pack.cpp" "-o" "/__w/llvm/llvm/build/SYCL/Reduction/Output/reduction_span_pack.cpp.tmp.out"
# command stderr:
warning: linking module '/__w/llvm/llvm/toolchain/lib/clang/15.0.0/../../clc/remangled-l64-signed_char.libspirv-amdgcn--amdhsa.bc': Linking two modules of different target triples: '/__w/llvm/llvm/toolchain/lib/clang/15.0.0/../../clc/remangled-l64-signed_char.libspirv-amdgcn--amdhsa.bc' is 'amdgcn-unknown-amdhsa' whereas '/__w/llvm/llvm/llvm_test_suite/SYCL/Reduction/reduction_span_pack.cpp' is 'amdgcn-amd-amdhsa'
 [-Wlinker-warnings]
1 warning generated.
$ ":" "RUN: at line 2"
note: command had no output on stdout or stderr
$ "true" "/__w/llvm/llvm/build/SYCL/Reduction/Output/reduction_span_pack.cpp.tmp.out"
note: command had no output on stdout or stderr
$ ":" "RUN: at line 3"
note: command had no output on stdout or stderr
$ "env" "SYCL_DEVICE_FILTER=ext_oneapi_hip:gpu,host" "/__w/llvm/llvm/build/SYCL/Reduction/Output/reduction_span_pack.cpp.tmp.out"
note: command had no output on stdout or stderr
error: command failed with exit status: 16
--
```

**To Reproduce**

This seems to be reliably reproducible in CI, but we've been unable to reproduce it on any machines we have access to. With the right machine, it should be possible to reproduce it by configuring with `--hip`, cloning the test suite from https://github.com/intel/llvm-test-suite, and running the test as described there.

**Environment (please complete the following information):**

- OS: Linux
- Target device and vendor: AMD Radeon RX 6700 XT 0.0
- DPC++ version: #6358
- Dependencies version: HIP 50221.15

**Additional context**

This has been blocking #6358 for over a week, but we've been unable to make any progress because the test output isn't very informative and we've been unable to find a system where we can reproduce it. It's been decided to mark this test XFAIL for now and continue with the pulldown while analysis of this bug is ongoing.

**Comment by zjin-lcf** [comment]
HI Daniel,

I tried to build the standalone test. The error is not reproducible on AMD MI-series GPUs.

Compiling the example takes long.

DPC++ 2022-06-release, ROCm 4.5.2

**Comment by dwoodwor-intel** [comment]
This problem seems to be very hardware-specific: @npmiller was also unable to reproduce this on an MI-series GPU in #6358 and I haven't been able to reproduce it on any machines with AMD GPUs I have access to either (though for most of them that might be more of a problem with not having the environment set up correctly).

Do you have access to any RX 6700 XTs like the one that was being used in CI where this was failing?

**Comment by zjin-lcf** [comment]
I see. When RX-series XT GPUs are available, I will try to run again.

**Comment by zjin-lcf** [comment]
I couldn't reproduce the error on a RX-series GPU after building/running the program.

**Comment by dwoodwor-intel** [comment]
It might be fixed now; I've opened intel/llvm-test-suite#1369 to try re-enabling the test

**Comment by dwoodwor-intel** [comment]
https://github.com/intel/llvm-test-suite/pull/1369 is merged (after being blocked for a little on some unrelated Windows test failures) and it seems like this test is passing again, so this issue can be closed
